ReactWindows/ReactNative/Modules/Storage/AsyncStorageModule.cs

```diff
@@ -20,5 +45,281 @@ public override string Name
+        /// <param name="callback">Callback.</param>
+        [ReactMethod]
+        public void getAllKeys(ICallback callback)
+        {
+            Debug.Assert(callback != null);
+
+            callback.Invoke(JToken.FromObject(_dataContainer.Values.Keys)); 
+        }
+
+        /// <summary>
+        /// Adds JToken to ApplicationDataContainer with the specified key.
+        /// </summary>
+        /// <param name="key">The key.</param>
+        /// <param name="token">The token.</param>
+        private void addTokenToContainer(string key, JToken token)
+        {
+            switch (token.Type)
+            {
+                case JTokenType.Null:
+                    _dataContainer.Values[key] = "";
+                    addTokenTypeToContainer(key, _DataType.Null);
+                    break;
+                case JTokenType.Boolean:
+                    _dataContainer.Values[key] = token.Value<bool>();
+                    break;
+                case JTokenType.Integer:
+                    _dataContainer.Values[key] = token.Value<long>();
+                    break;
+                case JTokenType.Float:
+                    _dataContainer.Values[key] = token.Value<double>();
+                    break;
+                case JTokenType.String:
+                    _dataContainer.Values[key] = token.Value<string>();
+                    addTokenTypeToContainer(key, _DataType.String);
+                    break;
+                case JTokenType.Array:
+                    _dataContainer.Values[key] = token.ToString();
+                    addTokenTypeToContainer(key, _DataType.Array);
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        /// <summary>
+        /// Gets related value for a specific key from ApplicationDataContainer as JToken.
+        /// </summary>
+        /// <param name="key">The key.</param>
+        private JToken getTokenFromContainer(string key)
+        {
+            var value = _dataContainer.Values[key];
+            if (value == null)
+            {
+                return JValue.CreateNull();
+            }
+            else
+            {
+                var t = value.GetType();
+                if (t == typeof(bool) || t == typeof(long) || t == typeof(double))
+                {
+                    return JToken.FromObject(value);
+                }
+                else if (t == typeof(string))
+                {
+                    var type = getTokenTypeFromContainer(key);
+                    if (type == _DataType.Array)
+                    {
+                        return JToken.Parse((string)value);
+                    }
+                    else if (type == _DataType.String)
+                    {
+                        return JToken.FromObject((string)value);
+                    }
+                    else
+                    {
+                        return JValue.CreateNull();
+                    }
```

Comment:
A switch case here on the enum types will be more maintainable going forward (if we add more types to put in the AsyncStorage).